### PR TITLE
start next Release (Version 4.2.0)

### PR DESCRIPTION
**Pull request by BeckerFrank**
following tasks are included:
 - update Mylyn Version from 4.1.0 to 4.2.0
 - update Mylyn Docs from 3.0.49 to 4.2.0
 - adjust API Baseline to 4.1.0
 - add Mylyn Docs projects to  API Baseline
 - reduce some Problem View Warnings
 - update mylyn.setup

### CODE REVIEW

**Review by ruspl-afed — APPROVED** [review]
Thank you so much for this effort @BeckerFrank !

**Comment by ruspl-afed** [comment]
> update Mylyn Docs from 3.0.49 to 4.2.0

Do you have any concerns regarding major version increment for Mylyn Docs @merks @akurtakov ?

**Comment by merks** [comment]
You've done it once before.  Did anyone complain?  ( Other than me.  😱 )

It's mostly a question of who is downstream from the bundle.  I only see this in SimRel where only a lower bound is specified:

image

**Review by merks — APPROVED** [review]
I cant' really look at 217 changed file sensibly, so I assume it's fine.

**Comment by akurtakov** [comment]
@leods I would defer to you the question about mylyn docs major version bump.

**Comment by ruspl-afed** [comment]
Thank you for the information regarding SimRel usages @merks 

> You've done it once before. Did anyone complain? ( Other than me. 😱 )

Yes, I did. 
And it was not my personal voluntary decision, we had an open poll [here](https://github.com/eclipse-mylyn/org.eclipse.mylyn/discussions/193)

**Comment by BeckerFrank** [comment]
> I cant' really look at 217 changed file sensibly, so I assume it's fine.

Should I split this in separate pull requests?

**Comment by merks** [comment]
> > I cant' really look at 217 changed file sensibly, so I assume it's fine.
> 
> Should I split this in separate pull requests?

No, we don't need to create more work for you.  The massive renumbering is just the way it is and we can assume it's fine.  The setup changes look fine too.  I'd like to get it all in so I can try it out more easily...

**Comment by BeckerFrank** [comment]
> > > I cant' really look at 217 changed file sensibly, so I assume it's fine.
> > 
> > 
> > Should I split this in separate pull requests?
> 
> No, we don't need to create more work for you. The massive renumbering is just the way it is and we can assume it's fine. The setup changes look fine too. I'd like to get it all in so I can try it out more easily...

I have created the branch 315-setup-test, which is already a stream "MylynDocs-Setup-Test" in the committed mylyn.setup, so you can test this immediately.

**Comment by merks** [comment]
It mostly looks good at the end:

image

**Comment by BeckerFrank** [comment]
> It mostly looks good at the end:
> 
> image

Yes, this is only due to the fact that the master version of mylyn.setup is used with API Baseline 3.26 at the first start. Please restart, then the API Baseline will be updated to 4.1.0 and the errors will be gone.

**Comment by BeckerFrank** [comment]
Commit [7db5e0f](https://github.com/eclipse-mylyn/org.eclipse.mylyn/pull/337/commits/7db5e0ff6e4bc0d10e6ebb31f45ca304ee880892) fix the mylyn.setup problem with the first startup

File: mylyn.context/org.eclipse.mylyn.cdt.tests/data/TestProject/.project [hunk 1]
```diff
@@ -5,23 +5,7 @@
 	<projects>
 	</projects>
 	<buildSpec>
-		<buildCommand>
```
Comment (gnl42) [inline mylyn.context/org.eclipse.mylyn.cdt.tests/data/TestProject/.project:8]: What was cdt doing in Mylyn?

**Review by gnl42 — APPROVED** [review]
looks good

File: mylyn.context/org.eclipse.mylyn.cdt.tests/data/TestProject/.project (hunk 1 above)
Comment (BeckerFrank) [reply mylyn.context/org.eclipse.mylyn.cdt.tests/data/TestProject/.project:8]: We have the org.eclipse.mylyn.cdt-feature feature!

**Comment by BeckerFrank** [comment]
> @leods I would defer to you the question about mylyn docs major version bump.

@leods: I wait with the merge of this pull request until I get the answer to the question.

**Comment by leods** [comment]
The version update for Docs is fine

**Pull request merged**